refactor(parser): store export entries in arena

camchenry opened the pull request:
I noticed we were using `vec![]` in the parser in just this one place. We probably don't want to do that, just to ensure everything is stored in the same place for caching.

camchenry commented:
* **#11567** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11567?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11567?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11567).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camchenry commented:
Nice, looks to actually be +1% on the Radix UI benchmark:
<img width="740" alt="Screenshot 2025-06-08 at 10 31 26 PM" src="https://github.com/user-attachments/assets/5c68bb28-e2f1-4fd1-a781-7ec13846fde9" />

Boshen commented:
### Merge activity

* **Jun 9, 3:56 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
refactor(parser): store export entries in arena (#11567)

I noticed we were using `vec![]` in the parser in just this one place. We probably don't want to do that, just to ensure everything is stored in the same place for caching.
